Wikipedia/Code/WMFExploreSectionControllerCache.m

```diff
@@ -43,22 +50,80 @@ - (instancetype)initWithDataStore:(MWKDataStore*)dataStore {
         self.dataStore                              = dataStore;
         self.sectionControllersBySection            = [[NSCache alloc] init];
         self.sectionControllersBySection.countLimit = [WMFExploreSection totalMaxNumberOfSections];
-        self.sectionsBySectionController            = [NSMapTable mapTableWithKeyOptions:NSMapTableWeakMemory | NSMapTableObjectPointerPersonality
-                                                                            valueOptions:NSMapTableWeakMemory];
+
+        /*
+         Controllers (keys in the map) are retained by the NSCache and can be evicted at any time. On the other hand,
+         sections (values in the map) are "value types" which should be owned by the receiver.  In other words, sections
+         are created as needed, and not retained for any particular reason, and it would be unintentional for a value in
```

Comment:
> I wasn't really taking that to heart. I'm against "weakly" referencing sections on principle, but in this case it doesn't matter since the weak map is just another "view" of the cache its associated with.

On second thought, it almost makes more sense to subclass `NSCache` and encapsulate this "reverse lookup" logic. At least for me, that would make this a less contentious.